opentaskspal/src/main/java/org/dmfs/opentaskspal/operations/Task.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2017 dmfs GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dmfs.opentaskspal.operations;
+
+import android.content.ContentProviderOperation;
+import android.support.annotation.NonNull;
+
+import org.dmfs.android.contentpal.InsertOperation;
+import org.dmfs.android.contentpal.Operation;
+import org.dmfs.android.contentpal.RowSnapshot;
+import org.dmfs.android.contentpal.SoftRowReference;
+import org.dmfs.android.contentpal.TransactionContext;
+import org.dmfs.android.contentpal.operations.Related;
+import org.dmfs.optional.Optional;
+import org.dmfs.tasks.contract.TaskContract;
+
+
+public final class Task implements InsertOperation<TaskContract.Tasks>
```

Comment:
The name is not ideal. This is a decorator which puts new a task on a specific list. The name should express that. Also it's valid to change the task list a list belongs to so this operation is not limited to inserts.

Reply:
This comes back to the question of whether to keep `InsertOperation` and `Operation`, because if we keep them, than we need to make 2 classes here for the same thing since it cannot be used in `Table.insertOpertation()` for return vallue if it's an `Operation`.
For now, I will just leave it as an `InsertOperation` and rename it.